Returned signatures of orders are sometimes too short

See the error of the end2end tests:

```
1) e2e
       Exchange
         #getOrderByHash()
           should return order by its hash:
      AssertionError: expected { Object (makerFee, takerFee, ...) } to deeply equal { Object (exchangeContractAddress, expirationUnixTimestampSec, ...) }
      + expected - actual
           "raw": "0"
           "value": "0"
         }
         "ecSignature": {
      -    "r": "0x2CA69B41318D36B921E2E3EAD9F5C0C5F459BF474C6B2E6642DB33424EF92E"
      +    "r": "0x002CA69B41318D36B921E2E3EAD9F5C0C5F459BF474C6B2E6642DB33424EF92E"
           "s": "0x1152506FE64DC305BE4035A6B57B735AACCF4F7EFDD47C9E73231081472D6005"
           "v": 28
         }
         "exchangeContractAddress": "0x731a10897d267e19B34503aD902d0A29173Ba4B1"
      
```

--- a/relay/api/exchange/resources.py
+++ b/relay/api/exchange/resources.py
@@ -6,36 +6,9 @@ from eth_utils import to_checksum_address, is_hex
 
 from relay.relay import TrustlinesRelay
 from relay.api import fields
+from relay.api.exchange.schemas import OrderSchema
 from relay.exchange.order import Order
 from relay.exchange.orderbook import OrderInvalidException
-
-
-def order_as_dict(order: Order):
-    return {
-        'exchangeContractAddress': order.exchange_address,
-        'maker': order.maker_address,
-        'taker': order.taker_address,
-        'makerTokenAddress': order.maker_token,
-        'takerTokenAddress': order.taker_token,
-        'feeRecipient': order.fee_recipient,
-        'makerTokenAmount': str(order.maker_token_amount),
-        'takerTokenAmount': str(order.taker_token_amount),
-        'filledMakerTokenAmount': str(order.filled_maker_token_amount),
-        'filledTakerTokenAmount': str(order.filled_taker_token_amount),
-        'cancelledMakerTokenAmount': str(order.cancelled_maker_token_amount),
-        'cancelledTakerTokenAmount': str(order.cancelled_taker_token_amount),
-        'availableMakerTokenAmount': str(order.available_maker_token_amount),
-        'availableTakerTokenAmount': str(order.available_taker_token_amount),
-        'makerFee': str(order.maker_fee),
-        'takerFee': str(order.taker_fee),
-        'expirationUnixTimestampSec': str(order.expiration_timestamp_in_sec),
-        'salt': str(order.salt),
-        'ecSignature': {
-            'v': int(order.v),
-            'r': '0x{:032X}'.format(int.from_bytes(order.r, 'big')),
-            's': '0x{:032X}'.format(int.from_bytes(order.s, 'big'))
-        }
-    }
 
 
 def abort_if_invalid_order_hash(order_hash):
@@ -58,11 +31,12 @@ class OrderBook(Resource):
         base_token_address = to_checksum_address(args['baseTokenAddress'])
         quote_token_address = to_checksum_address(args['quoteTokenAddress'])
         return {
-            'bids': [order_as_dict(order) for order in
-                     self.trustlines.orderbook.get_bids_by_tokenpair((base_token_address, quote_token_address))],
-            'asks': [order_as_dict(order) for order in
-                     self.trustlines.orderbook.get_asks_by_tokenpair((base_token_address, quote_token_address))],
-
+            'bids': OrderSchema().dump(
+                self.trustlines.orderbook.get_bids_by_tokenpair((base_token_address, quote_token_address)),
+                many=True),
+            'asks': OrderSchema().dump(
+                self.trustlines.orderbook.get_asks_by_tokenpair((base_token_address, quote_token_address)),
+                many=True),
         }
 
 
@@ -76,7 +50,7 @@ class OrderDetail(Resource):
         order = self.trustlines.orderbook.get_order_by_hash(bytes.fromhex(order_hash[2:]))
         if order is None:
             abort(404, message='Order does not exist')
-        return order_as_dict(order)
+        return OrderSchema().dump(order)
 
 
 class OrderSubmission(Resource):

--- a/relay/api/exchange/schemas.py
+++ b/relay/api/exchange/schemas.py
@@ -1,0 +1,36 @@
+from marshmallow import Schema, fields
+from ..fields import Address, BigInteger, HexBytes
+
+
+class SignatureSchema(Schema):
+    class Meta:
+        strict = True
+
+    v = fields.Integer()
+    r = HexBytes()
+    s = HexBytes()
+
+
+class OrderSchema(Schema):
+    class Meta:
+        strict = True
+
+    exchangeContractAddress = Address(attribute='exchange_address')
+    maker = Address(attribute='maker_address')
+    taker = Address(attribute='taker_address')
+    makerTokenAddress = Address(attribute='maker_token')
+    takerTokenAddress = Address(attribute='taker_token')
+    feeRecipient = Address(attribute='fee_recipient')
+    makerTokenAmount = BigInteger(attribute='maker_token_amount')
+    takerTokenAmount = BigInteger(attribute='taker_token_amount')
+    filledMakerTokenAmount = BigInteger(attribute='filled_maker_token_amount')
+    filledTakerTokenAmount = BigInteger(attribute='filled_taker_token_amount')
+    cancelledMakerTokenAmount = BigInteger(attribute='cancelled_maker_token_amount')
+    cancelledTakerTokenAmount = BigInteger(attribute='cancelled_taker_token_amount')
+    availableMakerTokenAmount = BigInteger(attribute='available_maker_token_amount')
+    availableTakerTokenAmount = BigInteger(attribute='available_taker_token_amount')
+    makerFee = BigInteger(attribute='maker_fee')
+    takerFee = BigInteger(attribute='taker_fee')
+    expirationUnixTimestampSec = BigInteger(attribute='expiration_timestamp_in_sec')
+    salt = BigInteger(attribute='salt')
+    ecSignature = fields.Nested(SignatureSchema, attribute='ec_signature')

--- a/relay/api/fields.py
+++ b/relay/api/fields.py
@@ -37,7 +37,7 @@ class BigInteger(fields.String):
 class HexBytes(fields.String):
 
     def _serialize(self, value, attr, obj):
-        return '0x{:032X}'.format(int.from_bytes(value, 'big'))
+        return '0x{:064X}'.format(int.from_bytes(value, 'big')).lower()
 
     def _deserialize(self, value, attr, data):
         value = super()._deserialize(value, attr, data)

--- a/relay/exchange/order.py
+++ b/relay/exchange/order.py
@@ -1,6 +1,11 @@
+from collections import namedtuple
+
 from eth_utils import is_checksum_address
 
 from tlcontracts.signing import keccak256, eth_sign, eth_validate
+
+
+EcSignature = namedtuple('EcSignature', 'v r s')
 
 
 class Order(object):
@@ -58,6 +63,10 @@ class Order(object):
     @property
     def available_taker_token_amount(self) -> float:
         return self.taker_token_amount - self.filled_taker_token_amount - self.cancelled_taker_token_amount
+
+    @property
+    def ec_signature(self):
+        return EcSignature(self.v, self.r, self. s)
 
     def validate(self) -> bool:
         return self.validate_signature() and self.validate_addresses()

